go/src/koding/klient/machine/machinegroup/mount.go

```diff
@@ -184,20 +206,74 @@ func (g *Group) AddMount(req *AddMountRequest) (res *AddMountResponse, err error
 		}
 	}()
 
+	addReq := &syncs.AddRequest{
+		MountID:       mountID,
+		Mount:         req.Mount,
+		NotifyBuilder: g.nb,
+		SyncBuilder:   g.sb,
+		ClientFunc:    g.dynamicClient(mountID),
+		SSHFunc:       g.dynamicSSH(req.ID),
+	}
+
 	// Start mount syncer.
-	dynSSH, dynClient := g.dynamicSSH(req.ID), g.dynamicClient(mountID)
-	if err = g.sync.Add(mountID, req.Mount, g.nb, g.sb, dynSSH, dynClient); err != nil {
+	if err = g.sync.Add(addReq); err != nil {
 		g.log.Error("Synchronization of %s mount failed: %s", mountID, err)
 		return nil, err
 	}
 
+	sc, err := g.sync.Sync(mountID)
+	if err != nil {
+		// Panic here since missing mount here should be impossible.
+		panic("mount " + req.Mount.String() + " doesn't exist")
```

Comment:
Five lines above we created a mount and new syncer(which we return here). The difference is that the only possible error returned from `g.sync` is `MountNotFound`, but it must be found since we just added it. Thus, we panic instead of returning error == the panic is for the sake of error handling.

UpdateIndex may receive mountID which doesn't exist, so the error is possible and logically valid.